apps/settings/js/panels/call_forwarding/panel.js

```diff
@@ -0,0 +1,270 @@
+define(function(require) {
+  'use strict';
+
+  var CallConstant = require('panels/call/call_constant');
+  var CallForwarding = require('panels/call/call_forwarding');
+  var CallUtils = require('panels/call/call_utils');
+  var DialogService = require('modules/dialog_service');
+  var SettingsPanel = require('modules/settings_panel');
+
+  return function ctor_call_barring() {
+    return SettingsPanel({
+      onInit: function(panel, options) {
+        this._elements = {};
+        this._elements.panel = panel;
+
+        this._cardIndex = options.cardIndex || 0;
+        this._conns = window.navigator.mozMobileConnections;
+        this._conn = this._conns[this._cardIndex];
+
+        // We will keep each callForwarding as internal variable
+        this._callForwardings = [null, null];
+
+        var $ = function(selector) {
+          return panel.querySelector(selector);
+        };
+
+        this._elements = {
+          liCfuDesc: $('#li-cfu-desc'),
+          liCfmbDesc: $('#li-cfmb-desc'),
+          liCfnRepDesc: $('#li-cfnrep-desc'),
+          liCfnReaDesc: $('#li-cfnrea-desc'),
+          cfuDesc: $('#cfu-desc'),
+          cfmbDesc: $('#cfmb-desc'),
+          cfnRepDesc: $('#cfnrep-desc'),
+          cfnReaDesc: $('#cfnrea-desc'),
+          mobileBusyMenuItem: $('.callForwardingMobileBusy'),
+          noReplyMenuItem: $('.callForwardingNoReply'),
+          notReachableMenuItem: $('.callForwardingNotReachable'),
+          unConditionalMenuItem: $('.callForwardingUnconditional')
+        };
+
+        this._cfOptionNameMapping = {
+          'unConditional': 'cfuDesc',
+          'mobileBusy': 'cfmbDesc',
+          'noReply': 'cfnRepDesc',
+          'notReachable': 'cfnReaDesc'
+        };
+
+        this._bindMenuItemsClickEvent();
+      },
+
+      onBeforeShow: function(panel, options) {
+        if (options.cardIndex) {
```

Comment:
If `cardIndex` is 0, it never enter this block. Please compare with `null` explicitly.